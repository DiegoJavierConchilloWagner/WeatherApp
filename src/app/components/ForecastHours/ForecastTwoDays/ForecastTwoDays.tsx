import { FC } from 'react';

import { useSelector } from 'react-redux';

import { ITwoDaysMapped } from '@common/interfaces';
import { RootState } from '@redux/store/store';

import { ForecastedHour } from '../ForecastedHour/ForecastedHour';
import { StyledForecastTwoDays } from '../styled';

interface Props {
	isLoading: boolean;
}

export const ForecastTwoDays: FC<Props> = ({ isLoading }) => {
	const { twoDays } = useSelector((state: RootState) => state.WeatherInterface);

	return (
		<StyledForecastTwoDays>
			{twoDays.map(
				(hour: ITwoDaysMapped, index: number) =>
					index > 0 && <ForecastedHour key={hour.dt} hour={hour} isLoading={isLoading} />
			)}
		</StyledForecastTwoDays>
	);
};
