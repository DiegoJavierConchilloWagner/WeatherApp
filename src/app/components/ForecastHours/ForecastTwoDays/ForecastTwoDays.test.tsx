import renderWithThemeAndProvider from '@common/utils/renderWhitThemeAndProvider';
import { ForecastTwoDays } from './ForecastTwoDays';

describe('Example ForecastTwoDays', () => {
	it('should render correcty', () => {
		const wrapper = renderWithThemeAndProvider(<ForecastTwoDays isLoading={false} />);

		expect(wrapper).toMatchSnapshot();
	});
});
