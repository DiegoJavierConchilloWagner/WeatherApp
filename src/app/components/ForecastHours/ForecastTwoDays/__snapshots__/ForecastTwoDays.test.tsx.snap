// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example ForecastTwoDays should render correcty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 3rem 10vw;
  background: rgba(255,255,255,0.1);
  box-shadow: 0 8px 32px 0 rgba(31,38,135,0.37);
  -webkit-backdrop-filter: blur(0px);
  backdrop-filter: blur(0px);
  -webkit-backdrop-filter: blur(0px);
  border-radius: 10px;
  border: 1px solid rgba(255,255,255,0.18);
}

@media (min-width:768px) {
  .c0 {
    overflow: auto;
    -webkit-scroll-snap-type: x mandatory;
    -moz-scroll-snap-type: x mandatory;
    -ms-scroll-snap-type: x mandatory;
    scroll-snap-type: x mandatory;
  }

  .c0::-webkit-scrollbar {
    height: 7px;
    width: 2px;
    background: #0d0f40;
  }

  .c0::-webkit-scrollbar-thumb:horizontal {
    background: #83c2da;
    border-radius: 25px;
  }
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    gap: 0%;
  }
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 3rem 10vw;
  background: rgba(255,255,255,0.1);
  box-shadow: 0 8px 32px 0 rgba(31,38,135,0.37);
  -webkit-backdrop-filter: blur(0px);
  backdrop-filter: blur(0px);
  -webkit-backdrop-filter: blur(0px);
  border-radius: 10px;
  border: 1px solid rgba(255,255,255,0.18);
}

@media (min-width:768px) {
  .c0 {
    overflow: auto;
    -webkit-scroll-snap-type: x mandatory;
    -moz-scroll-snap-type: x mandatory;
    -ms-scroll-snap-type: x mandatory;
    scroll-snap-type: x mandatory;
  }

  .c0::-webkit-scrollbar {
    height: 7px;
    width: 2px;
    background: #0d0f40;
  }

  .c0::-webkit-scrollbar-thumb:horizontal {
    background: #83c2da;
    border-radius: 25px;
  }
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    gap: 0%;
  }
}

<div>
    <div
      class="c0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
